### PR TITLE
[Datadog] add pagination to dashboard list

### PR DESCRIPTION
**Pull request by albertoxamin**
## Description

In larger organizations there can be thousands of dashboards 😅, using this extension with the non paginated api would just cause a out of memory exception.

## Screencast

from this 
image

to this
image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by kudrykv** [comment]
I'll check it out!

**Comment by pernielsentikaer** [comment]
Thanks for your contribution @albertoxamin 🔥 

Thanks for checking it out, awaiting your feedback @kudrykv 💪

**Comment by albertoxamin** [comment]
Wait I noticed this problem is also in the monitor list I'll push a fix for that too

**Comment by albertoxamin** [comment]
Ok now it's fixed!
image

File: extensions/datadog/src/command-list-dashboards.tsx [hunk 1]
```diff
@@ -1,27 +1,108 @@
...
+  });
+  const search = (query: string) => {
+    console.log("searching for dashboards", query);
+    setState(prev => ({
+      ...prev,
+      dashboardsAreLoading: true,
+    }));
+    fetch(
+      encodeURI(
+        `https://app.datadoghq.com/api/v1/dashboard_search?with_suggested=true&query=${query}&start=0&count=50&sort=`
```
Comment (kudrykv) [inline extensions/datadog/src/command-list-dashboards.tsx:53]: If the library has no query method and you must make the request on your own, you also need to account for the variable server.

File: extensions/datadog/src/command-list-dashboards.tsx [hunk 2]
```diff
@@ -1,27 +1,108 @@
-import { ActionPanel, List, OpenInBrowserAction } from "@raycast/api";
-import { useDashboards } from "./useDashboards";
+import { ActionPanel, List, OpenInBrowserAction, getPreferenceValues, Icon } from "@raycast/api";
 import { linkDomain } from "./util";
+import fetch from "node-fetch";
+import { useEffect, useState } from "react";
+
+export interface DashboardSummaryDefinition {
```
[the remark this answers is not in the document]
Comment (albertoxamin) [reply extensions/datadog/src/command-list-dashboards.tsx:6]: The dependency definition only has some of the fields

**Comment by kudrykv** [comment]
@albertoxamin, if you don't mind, I'll make a suggestion tomorrow on how we can improve this 🙂

**Comment by kudrykv** [comment]
@albertoxamin could you please add me to your fork, so I could commit there? Or do you want to cooperate in another way?
Also, I see that you work at Datadog. Wow 😲

**Comment by albertoxamin** [comment]
@kudrykv I added you. And yes I got to know about raycast and this extension from the company slack ahah.
Also they are hiring like crazy and there are teams extensively using React and also Go. If you are looking for a new job and need a referral let me know!

**Comment by kudrykv** [comment]
@albertoxamin, I've updated it; now you review it.
I'm thinking of how to improve listing monitors. What seems like "missing" to me is filtering out monitors by type (e.g., alerts-only) and by the team.

**Comment by albertoxamin** [comment]
@kudrykv Thanks for the update! I can't leave a code review on this repo, but it looks good and I can confirm your changes are working.
To me that is still an upgrade from just the standard filtering available in raycast
image
image
as it allows you to have very specific queries without much clicking around 
image

**Review by kudrykv — APPROVED** [review]

**Comment by kudrykv** [comment]
@pernielsentikaer please review :)

**Comment by pernielsentikaer** [comment]
Let's scratch that idea 👍

**Pull request merged**